source/common/upstream/outlier_detection_impl.cc

```diff
@@ -33,6 +33,21 @@ DetectorSharedPtr DetectorImplFactory::createForCluster(
   }
 }
 
+DetectorHostMonitorImpl::DetectorHostMonitorImpl(std::shared_ptr<DetectorImpl> detector,
+                                                 HostSharedPtr host)
+    : detector_(detector), host_(host),
+      // add Success Rate monitors
+      external_origin_SR_monitor_(std::make_unique<SuccessRateMonitor>(
```

Comment:
Can these just be member variables? Do they need to actually be pointers and allocated additionally?

Reply:
Done. Changed them to be member variables.